### PR TITLE
Add descriptive error messages #8

### PR DESCRIPTION
**Pull request by kizzanaome**
## Add descriptive error messages to the following fields
- username
- email
- password
- phone number.

### CODE REVIEW

File: apps/authentication/api/utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if re.search(r'(?=.*[\d])(?=.*[A-Z])(?=.*[a-z])', password) is None:
+        raise serializers.ValidationError("The password should have a capital letter, small letter and a number")
+    return password
+
+def validate_email(email):
+
+    check_email = User.objects.filter(email=email)
+    if check_email.exists():
+        raise serializers.ValidationError(f"A user with email {email} already exists.")
+    return email
```
Comment (kdaglas) [inline apps/authentication/api/utils.py]: Nice work Naome, I was thinking the email needs more validation than only this

Comment (kizzanaome) [reply apps/authentication/api/utils.py]: Thanks for the feed back however, most of the email validations are already catered for in the registration serializer class, on the part of extra_kwargs thats why i didnt add them here.

Comment (kdaglas) [reply apps/authentication/api/utils.py]: ok cool, great work 😊

**Comment by kdaglas** [comment]
[Use descriptive error messages and validate form data](https://trello.com/c/CS1zucR2/38-use-descriptive-error-messages-and-validate-form-data)

**Pull request merged**